packages/node_modules/@ciscospark/redux-module-media/src/actions.js

```diff
@@ -202,17 +202,29 @@ export function listenForIncomingCalls(sparkInstance, locusUrl) {
 
       return incomingCall.acknowledge()
         .then(() => {
-          const removeIncoming = () => dispatch(removeCall(incomingCall, incomingCall.locus.url));
+          const removeIncoming = () => {
+            incomingCall.off();
+            return dispatch(removeCall(incomingCall, incomingCall.locus.url));
+          };
 
           // If the remote party hangs up before we accept/decline
           incomingCall.once('inactive', removeIncoming);
 
-          // Once we are connected, our connected call state will handle disconnects
+          // If we rejected the call elsewhere, remove it
+          incomingCall.on('membership:declined', () => {
+            if (incomingCall.me.state === 'rejected') {
+              incomingCall.off('inactive', removeIncoming);
```

Comment:
Should we be keeping calls around until they become inactive? If remove them we can't maintain a list of active calls.

Reply:
True, instead of removing the call I'll do `declineIncomingCall` which marks the call as declined for recents.